### PR TITLE
Share scene assets from old scene when updating scene

### PR DESCRIPTION
**Pull request by tallytalwar**
Addresses #1473

### CODE REVIEW

File: core/src/scene/scene.cpp [hunk 1]
```diff
@@ -70,6 +71,9 @@ Scene::Scene(const Scene& _other)
     m_globalRefs = _other.m_globalRefs;
 
     m_mapProjection.reset(new MercatorProjection());
+    for (auto& sceneAsset: _other.sceneAssets()) {
+        m_sceneAssets[sceneAsset.first.k] = sceneAsset.second;
+    }
 }
```
Comment (hjanetzek) [inline core/src/scene/scene.cpp]: Any reason for not using `m_sceneAssets = _other.sceneAssets()` ?

File: core/src/scene/scene.h [hunk 2]
```diff
@@ -103,6 +105,9 @@ class Scene {
     const auto& featureSelection() const { return m_featureSelection; }
     const auto& sceneAssets() const { return m_sceneAssets; };
 
+    void createSceneAsset(const std::shared_ptr<Platform>& platform, const Url& resolvedUrl, const Url& relativeUrl,
+                          const Url& base);
+
```
Comment (hjanetzek) [inline core/src/scene/scene.h]: No need for this PR - I would rename these to`createAsset() / assets()` as it's clear that it's about parts of the scene

**Comment by tallytalwar** [comment]
There is also another bug I found in this. Pushing soon.

**Comment by tallytalwar** [comment]
Updated with a fix and scene urls are resolved post scene updates.

File: core/src/scene/scene.cpp (hunk 1 above)
Comment (tallytalwar) [inline core/src/scene/scene.cpp]: Ohh I think I had a different structure for scene assets before.. hence this. But make sense with `fastmap`. Updated.

File: core/src/scene/scene.h (hunk 2 above)
Comment (tallytalwar) [inline core/src/scene/scene.h]: Yeah make sense. Will push an explicit commit with this PR only. Thanks

**Pull request merged**